apps/src/gamelab/GameLabGroup.js

```diff
@@ -76,13 +76,28 @@ module.exports.Group = function (baseConstructor) {
    * @return {boolean} True if any touching occurred
    */
   array.isTouching = function (target) {
-    var didTouch = false;
-    for (var i = 0; i < this.length; i++) {
-      didTouch = this[i].isTouching(target) || didTouch;
-      // Note: can't early out because each call needs to update the sprite's
-      // touching property.
+    var state = jsInterpreter.getCurrentState();
```

Comment:
It's worth noting that this isn't just an interpreter-friendly for loop - it's interpreter-friendly `Array.prototype.reduce`.  Interpreter support for the actual `reduce` method was [added in June](https://github.com/NeilFraser/JS-Interpreter/commit/d80d8fb467aec9c443afd10f161eaeafee15ee86) and our copy of the interpreter is out-of-date and doesn't have this feature yet.  I think we should update our interpreter to get Neil's latest fixes, then see if we can implement this in terms of that `reduce` implementation.